src/Database/QueryCompiler.php

```diff
@@ -195,14 +185,17 @@ protected function _buildWithPart(array $parts, Query $query, ValueBinder $binde
      */
     protected function _buildSelectPart(array $parts, Query $query, ValueBinder $binder): string
     {
+        $driver = $query->getConnection()->getDriver($query->getConnectionRole());
         $select = 'SELECT%s %s%s';
-        if ($this->_orderedUnion && $query->clause('union')) {
+        if (
+            ($query->clause('union') || $query->clause('intersect')) &&
+            $driver->supports(DriverFeatureEnum::SET_OPERATIONS_ORDER_BY)
```

Comment:
@ajibarra This change is breaking UNION queries on mysql 5.7. The surrounding braces `(...)` are required even for mysql < 8.

Reply:
The `_orderedUnion` was default `true`, earlier the braces were added for all MySQL versions.

Reply:
Ok so the issue was the version set for order by in union but the braces should be added anyway...good to know...I see you already created a MR...Please let me know if there is anything I should help with here